foyer/smarts_graph.py

```diff
@@ -191,11 +191,10 @@ def find_matches(self, topology):
 
         # The first node in the smarts graph always corresponds to the atom
         # that we are trying to match.
-        first_atom = next(self.nodes_iter())
```

Comment:
@ctk3b Does this change make sense (along with the change to line 197/198)? This doesn't seem quite right, but all of our tests are passing...

Reply:
That might just be a vestige from an earlier time when we were still trying to use the actual `atom` objects as the node items. [These are now integers and the atom is the node attribute.](https://github.com/mosdef-hub/foyer/blob/master/foyer/smarts_graph.py#L53)

Easy check: set a breakpoint at 193 and see what nodes looks like. If it's just an integer then what you did is correct and I would move the comment in 192 down to 197.

Reply:
Yes, I checked this before I made the change and it seemed that `first_atom` was always an integer, specifically `0`. Sounds likely this change is okay then. I'll move that comment down.